lib/Doctrine/DBAL/Types/TypeRegistry.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\DBALException;
+use function array_search;
+use function in_array;
+
+/**
+ * Type registry is responsible for holding holding a map of known DBAL types.
+ * Type instances are implemented as flyweight pattern.
```

Comment:
```suggestion
 * Type instances implement the flyweight pattern.
```

Reply:
This is wrong, the types don't implement the pattern, they are used as such. Maybe using different word than ambiguous "implement" is a good idea.

Reply:
"Type instances are used as implementations of the flyweight pattern" then?